swig/include/python/gdal_python.i

```diff
@@ -3730,6 +3730,105 @@ def ApplyVerticalShiftGrid(*args, **kwargs):
     return _ApplyVerticalShiftGrid(*args, **kwargs)
 
 
+class VSIFile:
+
+    def __init__(self, path, mode, encoding="utf-8"):
+        self._path = path
+        self._mode = mode
+
+        self._binary = "b" in mode
+        self._encoding = encoding
+
+        self._fp = VSIFOpenL(self._path, self._mode)
+        self._closed = False
+        self._buffer = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        chunk_size = 1024
```

Comment:
probably that CPLReadLineL() could be used to avoid this reimplementatoin

Reply:
done